File: packages/lix/tests/forktree_stage2_coherent_view_audit/local_filesystem_hardcut_negative.ts
import { LocalFilesystem } from "@lix-js/sdk";

const positional = new LocalFilesystem("/tmp/workspace");
void positional;

// Every predecessor form must be unnameable. These directives make tsc fail
// if a removed API accidentally survives (unused @ts-expect-error).
// @ts-expect-error removed object/options constructor
new LocalFilesystem({ path: "/tmp/workspace", syncAllFiles: true });
// @ts-expect-error removed second constructor argument
new LocalFilesystem("/tmp/workspace", { syncAllFiles: false });
// @ts-expect-error removed public synchronization method
positional.importPaths(["note.md"]);
// @ts-expect-error removed public synchronization method
positional.syncDiskToLix();
// @ts-expect-error removed option property
positional.lixDir;
// @ts-expect-error removed option property
positional.syncAllFiles;

// @ts-expect-error removed exported option type
type RemovedOptions = import("@lix-js/sdk").LocalFilesystemOptions;
void (null as unknown as RemovedOptions);

File: packages/lix/tests/forktree_stage2_coherent_view_audit/local_filesystem_hardcut_negative.rs
//! Compile-fail input for the LocalFilesystem hard cut.
//!
//! The candidate qualifier compiles this as an external consumer and requires
//! unresolved import/associated-item diagnostics for every predecessor API.

use lix_storage_filesystem::{LocalFilesystem, LocalFilesystemOpenOptions};

fn old_api_must_not_compile() {
    let options = LocalFilesystemOpenOptions::new("/tmp/workspace", false);
    let _ = LocalFilesystem::open_with_options(options);
    let _ = LocalFilesystem::open_with_options_and_wasm_runtime;
    let _ = LocalFilesystem::import_paths::<Vec<&str>, &str>;
    let _ = LocalFilesystem::sync_disk_to_lix;
}

fn positive_signature_must_compile() {
    let _future = LocalFilesystem::open("/tmp/workspace");
}

fn main() {
    old_api_must_not_compile();
    positive_signature_must_compile();
}
